File: sdk/python/polar/customer_portal/organizations.py
from __future__ import annotations

from polar.base import AsyncServiceBase, SyncServiceBase, parse_response
from polar.errors import (
    HTTPValidationError,
    ResourceNotFound,
)
from polar.outputs import (
    CustomerOrganizationData,
)


class OrganizationsSync(SyncServiceBase):
    def get(
        self,
        slug: str,
    ) -> CustomerOrganizationData:
        """
        Get a customer portal's organization by slug.

        Args:
            slug: The organization slug.

        Raises:
            ResourceNotFound: Organization not found.
            HTTPValidationError: Validation Error
            PolarNetworkError: Raised when a network error occurs while making the request.
            PolarServerError: Raised when the server returns a 5xx error response.
        """
        request = self.client.build_request(
            method="GET",
            url="/v1/customer-portal/organizations/{slug}",
            path_params={
                "slug": slug,
            },
            query_params={},
        )
        response = self.client.send_request(request)
        method_errors = {
            404: ResourceNotFound,
            422: HTTPValidationError,
        }
        return parse_response(response, CustomerOrganizationData, method_errors)


class OrganizationsAsync(AsyncServiceBase):
    async def get(
        self,
        slug: str,
    ) -> CustomerOrganizationData:
        """
        Get a customer portal's organization by slug.

        Args:
            slug: The organization slug.

        Raises:
            ResourceNotFound: Organization not found.
            HTTPValidationError: Validation Error
        """
        request = self.client.build_request(
            method="GET",
            url="/v1/customer-portal/organizations/{slug}",
            path_params={
                "slug": slug,
            },
            query_params={},
        )
        response = await self.client.send_request(request)
        method_errors = {
            404: ResourceNotFound,
            422: HTTPValidationError,
        }
        return parse_response(response, CustomerOrganizationData, method_errors)

File: sdk/typescript/src/services/customer_portal/organizations.ts
import { ClientBase } from "../../base";
import type { CustomerOrganizationData } from "../../models/outputs";
import { HTTPValidationError, ResourceNotFound } from "../../errors";

export const getOrganizations = (client: ClientBase) => {
  /**
   * Get a customer portal's organization by slug.
   *
   * @param slug - The organization slug.
   * @returns {CustomerOrganizationData}
   * @throws {PolarNetworkError} When a network error occurs
   * @throws {PolarServerError} When the server returns a 5xx error
   * @throws {ResourceNotFound} Organization not found.
   * @throws {HTTPValidationError} Validation Error
   */
  return async (slug: string): Promise<CustomerOrganizationData> => {
    const pathParams = {
      slug: slug,
    };
    const queryParams = {};
    const request = client.buildRequest(
      "GET",
      "/v1/customer-portal/organizations/{slug}",
      pathParams,
      queryParams,
      undefined,
    );
    const response = await client.sendRequest(request);
    return client.parseResponse<CustomerOrganizationData>(response, "json", {
      404: ResourceNotFound,
      422: HTTPValidationError,
    });
  };
};

export function createOrganizationsService(client: ClientBase) {
  return {
    get: getOrganizations(client),
  };
}

export type Organizations = ReturnType<typeof createOrganizationsService>;
